.github/workflows/tests.yml

```diff
@@ -4,57 +4,118 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  IMAGE: appwrite-dev
+  CACHE_KEY: appwrite-dev-${{ github.event.pull_request.head.sha }}
```

Comment:
wouldn't this make every subsequent commit in PR use a different cache so we can't rely on docker image layer caching?

Reply:
This cache is created so it can be shared across different jobs in the same workflow. I did try 

As a result the image is just built once in the first step and its consumed by all the other jobs. Which is much faster than rebuilding the image in each step.

We still leverage docker image layer caching here https://github.com/appwrite/appwrite/blob/main/.github/workflows/tests.yml#L33-L34

The cache key I use is not related to the internal caches that docker uses